language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/docs/DocAttachmentInfo.java

```diff
@@ -87,20 +88,24 @@ public DocAttachmentInfo mergeDocAttachment(Documentation other) {
     }
 
     public String getDocumentationString() {
-        String result = String.format("# %s%n", this.description.trim());
+        StringBuilder result = new StringBuilder();
+        String[] descriptionLines = this.description.trim().split("\n");
```

Comment:
Shouldn't we use the platform specific line separator here?

Reply:
@nadeeshaan  Actually in this commit: https://github.com/ballerina-platform/ballerina-lang/pull/29565/commits/85e91526d20b8fdb8e3972641ad7bdb059cdb806, I had the `System.lineSeparator()` for splitting. However, this caused the windows unit test for `UpdateDocumentationCommandExecutorTest` to fail. Assuming that this was due to the `\r\n` in windows, I changed it to `\n` as now, and the windows test passed afterwards. It's a bit confusing :thinking:

Reply:
Updated again to use `System.lineSeparator()`. Let's see what happens now.

Reply:
If these are still failing, I think our tests are not aware of the platform risgt?

Reply:
Seems like the `description` in `DocAttachmentInfo` doesn't contain platform specific line separators. This cause the tests to fail on windows when `System.lineSeparator()` is used because the `description` is separated only with `\n`. For now, we are going to use `\n` for separating lines as discussed via zoom.